psalm-baseline.xml

```diff
@@ -114,10 +114,16 @@
     <InvalidParamDefault>
       <code><![CDATA[class-string<ArrayIterator<string,mixed>>|class-string<ArrayObject<string,mixed>>]]></code>
     </InvalidParamDefault>
+    <LessSpecificReturnStatement>
+      <code><![CDATA[(object) $this->getArrayCopy()]]></code>
+    </LessSpecificReturnStatement>
     <MixedAssignment>
       <code>$this[$key]</code>
       <code>$value</code>
     </MixedAssignment>
+    <MoreSpecificReturnType>
+      <code>stdClass</code>
+    </MoreSpecificReturnType>
```

Comment:
The `LessSpecificReturnStatement` and `MoreSpecificReturnType` are due to https://github.com/vimeo/psalm/issues/8187. In essence, Psalm takes the return value of `(object) []` to be `object`, even though it is `stdClass`.